src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -266,7 +266,7 @@ private static IntPtr TryGetThunkDataAddress(IntPtr thunkAddress)
             nuint thunkAddressValue = (nuint)(nint)ClearThumbBit(thunkAddress);
 
             // Compute the base address of the thunk's mapping
-            nuint currentThunksBlockAddress = thunkAddressValue & ~Constants.ThunkBlockSizeMask;
+            nuint currentThunksBlockAddress = thunkAddressValue & ~Constants.ThunkDataBlockSizeMask;
```

Comment:
We are applying DataBlockSizeMask to code address. Is that going to work in all cases?

Reply:
Yes, before https://github.com/dotnet/runtime/pull/88710 this was page size before and each block is considered to be a page size. To be more pure we could add a corresponding function to return the size of the code block, but it will be the same size as the data block.